finance/portfolio-performance/Portfile

```
@@ -0,0 +1,41 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        buchen portfolio 0.49.4
+name                portfolio-performance
+categories          finance
+platforms           darwin
+license             EPL-1
+maintainers         @janosch mailbox.org:janosch1 openmaintainer
+description         A simple tool to calculate the overall performance of an investment portfolio.
+long_description    An open source tool to calculate the overall performance of an investment portfolio \
+                    - across all accounts - using True-Time Weighted Return or Internal Rate of Return.
+homepage            https://www.portfolio-performance.info/en/
+
+depends_build       port:maven3 \
```

```suggestion
depends_build       bin:mvn3:maven3 \
```

Only the binary is required as part of the build process, so I think it should be safe to specify it as a file dependency.

See https://guide.macports.org/chunked/reference.dependencies.html for more details.

---

AFAIK binary dependecies should be the exception, no? From the guide:

> But when satisfying a dependency with vendor-supplied software is preferred for special reasons, or when it is possible for more than one port to satisfy a dependency, then file dependencies may be used. 

Neither of those two mentioned cases apply in this situation, so I thought a port dependency makes more sense. No?

---

Thanks @Janosch for your reply. I'd say that it's the second point:

> it is possible for more than one port to satisfy a dependency

There are many different maven ports that provide the required binary (e.g. [maven31](https://github.com/macports/macports-ports/blob/749ad2787c30497020bdc4cca116eff9f09c800b/java/maven31/Portfile), [maven32](https://github.com/macports/macports-ports/blob/749ad2787c30497020bdc4cca116eff9f09c800b/java/maven32/Portfile), and of course, [maven3](https://github.com/macports/macports-ports/blob/749ad2787c30497020bdc4cca116eff9f09c800b/java/maven3/Portfile)).

---

Are you sure? Ony my machine it looks like this when I install maven32 and maven31 (and do the `port select --set maven`):
```
~/Dev/macports-ports ❯❯❯ ls -lah /opt/local/bin/mvn*
lrwxr-xr-x  1 root  wheel    20B 24 Jan 13:04 /opt/local/bin/mvn@ -> /opt/local/bin/mvn31
lrwxr-xr-x  1 root  admin    40B 24 Jan 13:04 /opt/local/bin/mvn31@ -> ../share/java/apache-maven-3.1.1/bin/mvn
lrwxr-xr-x  1 root  admin    29B 24 Jan 13:00 /opt/local/bin/mvn32@ -> ../share/java/maven32/bin/mvn
lrwxr-xr-x  1 root  wheel    25B 24 Jan 13:04 /opt/local/bin/mvnDebug@ -> /opt/local/bin/mvnDebug31
lrwxr-xr-x  1 root  admin    45B 24 Jan 13:04 /opt/local/bin/mvnDebug31@ -> ../share/java/apache-maven-3.1.1/bin/mvnDebug
lrwxr-xr-x  1 root  admin    34B 24 Jan 13:00 /opt/local/bin/mvnDebug32@ -> ../share/java/maven32/bin/mvnDebug
~/Dev/macports-ports ❯❯❯ mvn3
fish: Unknown command: mvn3
```
IMHO, only the port maven3 provides the binary `mvn3`. A different story would be the binary (or symlink)`mvn`. But in my case I need maven 3.x. so I cannot just use any mvn binary.

I can still change it if you insist but it feels like violating the macports guidelines.

---

I see. Thanks for testing that out, I didn't appreciate that although the `mvn` binary is provided, the `mvn3` binary isn't. Who would've known that? (You @Janosch, that's who 👍)

After doing some research, it seems that prior to Mavericks (10.9), maven3 came preinstalled. This might lead to the scenario "where the versions in macOS are often sufficient", but I don't know whether the `mvn3` binary is present or not.